### DISCUSSION TITLE
Browserstack Automate - HTTP keep-alive missing

### DISCUSSION THREAD

**Discussion by user7699** [webdriverio/webdriverio#11408] [Q&A]
Whenever I use Browserstack Automate I keep getting this error on every test in the Browserstack portal: 

**HTTP keep-alive missing**
_Session did not have HTTP keep-alive directive when connecting to the BrowserStack cloud. You might have experienced slow performance in this session. Read our documentation to learn how to use keep-alive._

A reference on how to fix the error: https://www.browserstack.com/docs/automate/selenium/error-codes/keep-alive-not-used

![bild](https://github.com/webdriverio/webdriverio/assets/93139584/f8945445-76e8-4a8d-b9f7-63afbd0ea741)

I'm guessin this error stems from me using a proxy to access the website that I'm testing. Here are my capabilities: 

        "@types/mocha": "^10.0.1",
        "@types/node": "^20.2.5",
        "@wdio/browserstack-service": "^8.17.0",
        "@wdio/cli": "^8.17.0",
        "@wdio/local-runner": "^8.17.0",
        "@wdio/mocha-framework": "^8.17.0",
        "@wdio/spec-reporter": "^8.17.0",
         "global-agent": "^3.0.0",

Where global-agent is the package used for setting up the proxy, in my config-file:

require('global-agent/bootstrap');
global.GLOBAL_AGENT.HTTP_PROXY = 'http://xx.xx.xx.xx:xxxx/';

The referenced fix that Browserstack is providing doesn't work for me.

 Does anyone have a fix for this? I'm guessing I might have to set up my proxy in a different way if global-agent doesn't provide a way for me to use "keep-alive"

**Comment by grandEL-dsouza** [comment]
hey, 

Please check which network calls are specifically responsible for the delay here. You can then perhaps check the server logs or optimise the same at your end. 

Additionally run a few tests by disabling the network logs on Browserstack because Browserstack introduces an MITM proxy to capture the network logs. The proxy you configured and the MITM proxy could be adding up to the delay. 

I hope this helps you. 

Thanks

**Comment by user7699** [comment]
Thank you @grandEL-dsouza, disabling network logs didn't help but in the network logs on the Browserstack Automate portal I could see that maybe half of the requests had keep-alive: true. I will have to dig further into this problem.
